core/src/test/java/org/apache/iceberg/TestDataTableScan.java

```diff
@@ -27,4 +34,56 @@ public TestDataTableScan(int formatVersion) {
   protected TableScan newScan() {
     return table.newScan();
   }
+
+  @Test
+  public void testTaskRowCounts() {
+    Assume.assumeTrue(formatVersion == 2);
+
+    DataFile dataFile1 = newDataFile("data_bucket=0");
+    table.newFastAppend().appendFile(dataFile1).commit();
+
+    DataFile dataFile2 = newDataFile("data_bucket=1");
+    table.newFastAppend().appendFile(dataFile2).commit();
+
+    DeleteFile deleteFile1 = newDeleteFile("data_bucket=0");
+    table.newRowDelta().addDeletes(deleteFile1).commit();
+
+    DeleteFile deleteFile2 = newDeleteFile("data_bucket=1");
+    table.newRowDelta().addDeletes(deleteFile2).commit();
+
+    TableScan scan = table.newScan().option(TableProperties.SPLIT_SIZE, String.valueOf(50));
```

Comment:
nit: is `"50"` simpler than `String.valueOf(50)`?

Reply:
Yeah, I think I copied it from somewhere else. I'll change.

Reply:
Done.